app/models/workflow.rb

```diff
@@ -6,6 +6,8 @@ class Workflow < ApplicationRecord
 
   validates :name, :presence => :name
 
+  before_destroy :not_to_delete_default
```

Comment:
Overall, this looks good but suggest we use a different method name.  Maybe `protect_default`